### PR TITLE
Update README versions with badge

### LINKED ISSUE #372: Update README
README is not up-to-date.

### PR DESCRIPTION
**Pull request by shaunkawano**
close #372

### CODE REVIEW

**Comment by Tarrasch** [comment]
Yea. I had to waste quite some time yesterday due to confusion caused by this. At some point I thought you were supposed to use `dagger-2.2` with `dagger-compiler-2.4`. That said I'm a Java noob but I think this readme ought to cater to everyone. :)

Another idea. Isn't it better to just link to maven central or alternatively use some kind of badge? Like this one: 

[Maven Central](https://maven-badges.herokuapp.com/maven-central/com.google.dagger/dagger).

`[Maven Central](https://maven-badges.herokuapp.com/maven-central/com.google.dagger/dagger)` (copy paste this)

**Comment by shaunkawano** [comment]
@Tarrasch 

Thanks.

> At some point I thought you were supposed to use dagger-2.2 with dagger-compiler-2.4

I did not know that. Is it still applicable?

The badge looks nice; I'd like to hear from the guys at Google if the badge is a good idea.

**Comment by Tarrasch** [comment]
> I did not know that. Is it still applicable?

No no. I just meant that *I got confused*. Afaict you always use the same version number on all packages at one given time, like `dagger-2.4` with `dagger-compiler-2.4`.

> The badge looks nice; I'd like to hear from the guys at Google if the badge is a good idea.

Yea. It seems like exactly what we want here. There's also this shields version: http://shields.io/ which has even shorter url and looks nicer.

And thanks to you too @shaunkawano  :)

**Comment by gk5885** [comment]
I had no idea that the badge was a thing.  That's pretty nifty.  I'd be fine with just using using the badge. @ronshapiro or @netdpb have any objections?

**Comment by shaunkawano** [comment]
@gk5885 Thanks for your response. I'll add maven central badge then.

I have also thought that maybe we can remove all specific library version numbers from README as well so that you do not have to update README just to modify the numbers.

**Comment by Tarrasch** [comment]
> I have also thought that maybe we can remove all specific library version numbers from README as well so that you do not have to update README just to modify the numbers.

Yea, that kind of information always goes out of date.

**Comment by AlokBansal8** [comment]
👍

**Comment by shaunkawano** [comment]
I'm not necessarily willing MY PR to be merged, but I believe README should always be up-to-date otherwise it may confuse people? 👀

**Comment by AlokBansal8** [comment]
I support that the README should always be updated. I would also love to
see the apt and compile code for gradle and not just about jar.

On Sat 28 May, 2016, 11:29 AM Shaun Kawano, <notifications@github.com>
wrote:

> I'm not necessarily willing MY PR to be merged, but I believe README
> should always be up-to-date otherwise it may confuse people? 👀
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/dagger/pull/373#issuecomment-222292140>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ALfOHWB2BZpo7y7z6oLzH1627YO8RjXnks5qF9nXgaJpZM4IZUkr>
> .
>
-- 
-Thanks,
Alok Bansal

**Comment by ersin-ertan** [comment]
http://google.github.io/dagger/users-guide.html
Users guide documentation will have the same problem.

**Comment by shaunkawano** [comment]
Any updates? 👀

**Comment by ersin-ertan** [comment]
I think [this solution by Google (embedmd)](https://github.com/campoy/embedmd) could reference the version code and have synced in all of the documentation.

**Comment by shaunkawano** [comment]
Almost 3 months passed; So ironic that the latest version 2.6 came along and the README still indicates that `2.2` is the latest stable release. 😂

**Comment by cgruber** [comment]
I'm could you please rebase this?   I was about to update the readme and @ronshapiro reminded me that you had done this work.

**Comment by cgruber** [comment]
Also, kudos on pointing out the badges. That's way better than manually keeping track on our end.

**Comment by shaunkawano** [comment]
@cgruber Sorry took me some time to rebase but I think and hope I've done it correctly. Could you check my commit again whenever you are free? Thank you.

**Comment by cgruber** [comment]
Cool.  Its been merged internally and should make it out on the next sync.
We'll close it then.

File: README.md [hunk 1]
```diff
@@ -17,16 +19,6 @@ Square, Google, and other contributors.
...
-## Status
-
-  - ***Release Version:* 2.2**
-  - ***Snapshot Version:* 2.3-SNAPSHOT**
-
-Dagger is currently in active development, primarily internally at Google,
-with regular pushes to the open-source community. Snapshot releases are
-auto-deployed to sonatype's central maven repository on a clean build with
-the version `2.3-SNAPSHOT`.
-
```
Comment (Tarrasch) [inline README.md]: Removing cruft from the README. Awesome!

Comment (shaunkawano) [inline README.md]: @Tarrasch I know right?
 
Good job Pikachu!

image

**Comment by cgruber** [comment]
All merged in.  Thanks!

**Pull request merged**